### PR TITLE
Add wasm-3.0 branch to the sync'd tests here

### PR DESCRIPTION
**Pull request by alexcrichton**
I've tried to hack this into update-testsuite.sh but I probably made a mistake here or there. I believe this handles the removed tests in #109 though.

### CODE REVIEW

File: update-testsuite.sh [hunk 1]
```diff
@@ -165,10 +173,14 @@ for repo in ${repos}; do
...
-        log_and_run git add ${wast_dir}/*.wast
-
-        repo_sha=$(git -C repos/${repo} log --max-count=1 --oneline origin/HEAD| sed -e 's/ .*//')
-        echo "  ${repo}:" >> commit_message
+        log_and_run git add ${wast_dir}
+
+        branch=main
+        dir=${repo}
+        [ "${repo}" == "wasm-3.0" ] && branch="wasm-3.0"
+        [ "${repo}" == "wasm-3.0" ] && repo="spec"
```
Comment (sbc100) [inline update-testsuite.sh:181]: Maybe use a multiline `if` block here?

```
if [ "${repo}" == "wasm-3.0" ]; then
  branch="wasm-3.0"
  repo="spec"
fi
```

File: update-testsuite.sh [hunk 2]
```diff
@@ -122,6 +129,7 @@ for repo in ${repos}; do
         continue
     fi
 
+    echo "++ updating222 ${repo}"
```
Comment (sbc100) [inline update-testsuite.sh:132]: Debugging?

File: update-testsuite.sh [hunk 3]
```diff
@@ -80,6 +85,8 @@ merge_with_spec() {
 
     set_upstream ${repo}
 
+    [ "${repo}" == "wasm-3.0" ] && return
```
Comment (sbc100) [inline update-testsuite.sh:88]: Can you add a comment here describing why we early return in this case?

File: update-testsuite.sh [hunk 4]
```diff
@@ -44,18 +45,22 @@ popdir() {
 }
 
 update_repo() {
-    local repo=$1
+    local dir=$1
+    local branch=main
+    local repo=$dir
+    [ "${repo}" == "wasm-3.0" ] && branch="wasm-3.0"
+    [ "${repo}" == "wasm-3.0" ] && repo="spec"
```
Comment (sbc100) [inline update-testsuite.sh:52]: Multiline if block like below.

**Review by sbc100 — COMMENTED** [review]
I'm curious why the removal from #109 are not also part of this change?

File: update-testsuite.sh (hunk 3 above)
Comment (alexcrichton) [reply update-testsuite.sh]: In commenting this I realized it was actually easier to just go ahead and handle this, so it no longer early-returns

**Review by sbc100 — APPROVED** [review]
Ah I see.  Thanks for the explanation.

**Review by sbc100 — COMMENTED** [review]
I think we are approaching max-complexity for bash scripts here.. maybe we should convert this to python at some point in the future?

**Comment by alexcrichton** [comment]
I felt like giving it a spin so I've pushed a rewrite from bash to python in this PR as well. I'm happy to split that out to a separate PR if preferred. I've tested it locally by ensuring that if I run it it does nothing (e.g. no updates) and then if I revert the updates made in this PR it generates the same set of updates.

One major change from before is that it no longer runs `git clone` or manages a git repository for each upstream proposal. I found that made the script quite slow the first time it was run when internet wasn't fast because the spec repo is relatively large. Now there's just a single git checkout and everything is fetched into that so shared history should only be fetched at most once. Additionally only a single branch is fetched which should avoid binary artifacts on other branches.

**Comment by alexcrichton** [comment]
Sounds good, rolled back that commit and I'll post it after this merges

**Comment by alexcrichton** [comment]
Oh I'll interpret @sbc100 your approval as "ok for me to merge" so I'm going to merge. (sorry I forget which projects prefer others merge vs which are ok with self-merging)

**Pull request merged**